components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -815,40 +821,100 @@ public Response getAmazonResourceNamesOfAPI(String apiId, MessageContext message
-                                    !StringUtils.isEmpty(region)) {
+                        String roleArn = (String) endpointConfig.get(APIConstants.AMZN_ROLE_ARN);
+                        String roleSessionName = (String) endpointConfig.get(APIConstants.AMZN_ROLE_SESSION_NAME);
+                        String roleRegion = (String) endpointConfig.get(APIConstants.AMZN_ROLE_REGION);
+                        AWSLambda awsLambdaClient;
+                        if (StringUtils.isEmpty(accessKey) && StringUtils.isEmpty(secretKey)) {
+                            if (log.isDebugEnabled()) {
+                                log.debug("Using temporary credentials supplied by the IAM role attached to AWS " +
+                                        "instance");
+                            }
+                            if (StringUtils.isEmpty(roleArn) && StringUtils.isEmpty(roleSessionName)
+                                    && StringUtils.isEmpty(roleRegion)) {
+                                awsLambdaClient = AWSLambdaClientBuilder.standard()
+                                        .withCredentials(DefaultAWSCredentialsProviderChain.getInstance())
+                                        .build();
+                            } else if (StringUtils.isNotEmpty(roleArn) && StringUtils.isNotEmpty(roleSessionName)
+                                    && StringUtils.isNotEmpty(roleRegion)) {
+                                AWSSecurityTokenService awsSTSClient = AWSSecurityTokenServiceClientBuilder.standard()
+                                        .withCredentials(DefaultAWSCredentialsProviderChain.getInstance())
+                                        .build();
+                                AssumeRoleRequest roleRequest = new AssumeRoleRequest()
+                                        .withRoleArn(roleArn)
+                                        .withRoleSessionName(roleSessionName);
+                                AssumeRoleResult assumeRoleResult = awsSTSClient.assumeRole(roleRequest);
+                                Credentials sessionCredentials = assumeRoleResult.getCredentials();
+                                BasicSessionCredentials basicSessionCredentials = new BasicSessionCredentials(
+                                        sessionCredentials.getAccessKeyId(),
+                                        sessionCredentials.getSecretAccessKey(),
+                                        sessionCredentials.getSessionToken());
+                                awsLambdaClient = AWSLambdaClientBuilder.standard()
+                                        .withCredentials(new AWSStaticCredentialsProvider(basicSessionCredentials))
+                                        .withRegion(roleRegion)
+                                        .build();
+                            } else {
+                                log.error("Missing AWS STS configurations");
```

Comment:
can u explain the reason behind login the error only and ignoring the it. how this faulty case is handled after that?